### DISCUSSION TITLE
B-table header-checkable doesn't have the same is-checked property as the checkboxes in the rows?

### DISCUSSION THREAD

**Discussion by liliannemueller** [buefy/buefy#4054] [Q&A]
So when a checkbox in a b-table is checked, the class `"is-checked"` is added to the element. Is there a different property that gets added to the header checkbox? I am using the "is-checked" class to style my checkboxes dynamically but am unable to apply the same style to my header checkbox. Please help!

**Comment by wesdevpro** [comment]
> So when a checkbox in a b-table is checked, the class `"is-checked"` is added to the element. Is there a different property that gets added to the header checkbox? I am using the "is-checked" class to style my checkboxes dynamically but am unable to apply the same style to my header checkbox. Please help!

@liliannemueller Excellent question! There may be a styling conflict occurring in the background. I'll need to dive deeper into the documentation and source code to determine if there's a clear reason why the 'is-checked' style isn't applying. Could you provide a code example of how you're attempting to apply the style? Also, what version of Buefy/Vue are you using?

**Reply by liliannemueller** [reply]
Vue version: 2.6.14
Buefy version: 0.9.21

So I'm able to apply the styling to my row checkboxes like this:  
```
tr.is-checked .b-checkbox::after {
            opacity: 1;  // Shows checkmark
        }
```

When I try to do the same for my header, nothing happens.
```
.th .b-checkbox.is-checked{
                opacity: 1;
            }
```
When I inspect with dev tools, I can clearly see `is-checked` being toggled on/off for the rows, but nothing on the header.

**Reply by wesdevpro** [reply]
Can you see the CSS style for the headers in the dev tools? Depending on your HTML you may be incorrectly selecting the header check boxes. Try adding an absurd color or background color to see if it applies to those checkboxes

**Reply by liliannemueller** [reply]
`<th class="checkbox-cell"><label class="b-checkbox checkbox"><input type="checkbox" autocomplete="off" true-value="true" value="false"><span class="check is-primary"></span><span class="control-label"></span></label></th>`
Here's the header element when checked!

**Reply by wesdevpro** [reply]
This is probably not related but does HTML5 require the input tag to end in a `/>`
```html
....<input type="checkbox" autocomplete="off" true-value="true" value="false" />....
```

**Reply by wesdevpro** [reply]
> Vue version: 2.6.14 Buefy version: 0.9.21
> 
> So I'm able to apply the styling to my row checkboxes like this:
> 
> ```
> tr.is-checked .b-checkbox::after {
>             opacity: 1;  // Shows checkmark
>         }
> ```
> 
> When I try to do the same for my header, nothing happens.
> 
> ```
> .th .b-checkbox.is-checked{
>                 opacity: 1;
>             }
> ```
> 
> When I inspect with dev tools, I can clearly see `is-checked` being toggled on/off for the rows, but nothing on the header.

@liliannemueller you are selecting `th` as a class and not an element...

Your CSS should be as follows:
```css
th.b-checkbox.is-checked {
    opacity: 1;
 }
 ```

**Comment by liliannemueller** [comment]
Sorry, that was my typo, I do have 

```
th.b-checkbox.is-checked {
        opacity: 1;
     }
```

but it's still not working.

**Reply by liliannemueller** [reply]
is `is-checked`` also supposed to be applied to the header checkbox when checked?

**Reply by wesdevpro** [reply]
> Sorry, that was my typo, I do have
> 
> ```
> th.b-checkbox.is-checked {
>         opacity: 1;
>      }
> ```
> 
> but it's still not working.

No worries.

**Reply by wesdevpro** [reply]
> is `is-checked`` also supposed to be applied to the header checkbox when checked?

Try removing the .is-checked from your CSS selector

**Reply by liliannemueller** [reply]
I'm able to apply general styling to the header checkbox with `th .b-checkbox`, but I want to change the styling when it gets checked, so I'm trying to figure out how to get the `is-checked` class to show up on the header checkbox like it does on the row checkboxes.

**Reply by wesdevpro** [reply]
> I'm able to apply general styling to the header checkbox with `th .b-checkbox`, but I want to change the styling when it gets checked, so I'm trying to figure out how to get the `is-checked` class to show up on the header checkbox like it does on the row checkboxes.

You are on the right track. Keep your CSS style defined but on top of the CSS you need to add some vue JavaScript to toggle the CSS is-checked class on whatever element you want to apply that `opacity: 1;`

**Reply by wesdevpro** [reply]
> I'm able to apply general styling to the header checkbox with `th .b-checkbox`, but I want to change the styling when it gets checked, so I'm trying to figure out how to get the `is-checked` class to show up on the header checkbox like it does on the row checkboxes.

As a test try manually adding the is-checked class to the element you are trying to apply the CSS class too

**Reply by wesdevpro** [reply]
> > I'm able to apply general styling to the header checkbox with `th .b-checkbox`, but I want to change the styling when it gets checked, so I'm trying to figure out how to get the `is-checked` class to show up on the header checkbox like it does on the row checkboxes.
> 
> You are on the right track. Keep your CSS style defined but on top of the CSS you need to add some vue JavaScript to toggle the CSS is-checked class on whatever element you want to apply that `opacity: 1;`

You can try one of these events to create logic to toggle the is-checked class
![image](https://github.com/user-attachments/assets/d84da1b9-dcbb-45a3-8e92-252e277035bb)

**Reply by wesdevpro** [reply]
@liliannemueller here is an AI-generated example. It is not specific to Buefy but the logic is the same.

Note: there are some other ways to do this.

```vuejs
<template>
  <div>
    <button @click="toggleClass" :class="{ active: isActive }">Toggle Class</button>
  </div>
</template>

<script>
export default {
  data() {
    return {
      isActive: false
    };
  },
  methods: {
    toggleClass() {
      this.isActive = !this.isActive;
    }
  }
};
</script>

<style>
.active {
  background-color: yellow;
}
</style>
```

**Reply by liliannemueller** [reply]
Okay cool, thanks for your advice... I will give this a shot!

**Reply by wesdevpro** [reply]
> Okay cool, thanks for your advice... I will give this a shot!

Sweet!

If you need any further assistance, please let me know! You can reach me via the discord up until 10:00 pm EST. And if necessary, we can do a screen share.

**Reply by liliannemueller** [reply]
Awesome, thanks so much!

**Reply by liliannemueller** [reply]
No luck with this approach. Wondering if there is a different way to target a specific header cell with the `:header-class` property?

**Reply by wesdevpro** [reply]
> header-class

@liliannemueller please see https://github.com/buefy/buefy/issues/2414

If you still require a screen share I will be available around 9:30 EST
